### DISCUSSION TITLE
Use Open SSH for local (file://) remote repositories, when Putty is selected

### DISCUSSION THREAD

**Discussion by regs01** [gitextensions/gitextensions#10576] [Q&A]
### Environment

- Git Extensions 4.0.1.15887
- Build f2567dea28bbf0e2cbcd2a95d4c7200e47d0d671
- Git 2.39.0.windows.2
- Microsoft Windows NT 10.0.19044.0
- .NET 6.0.12
- DPI 144dpi (150% scaling)

### Issue description

Putty doesn't work with local remote repositories (like g:\git or file://g:/git). So when you have both remote remote and local remote in your project and using putty, local stops working (insisting on key or unable to resolve host).

### Steps to reproduce

- Add SSH or HTTPS remote repository.
- Add folder as remote repository.
- Switch to Putty
- SSH or HTTPS are working, but FILE isn't

**Comment by vbjay** [comment]
Doubtful putty is involved with that.

**Comment by regs01** [comment]
<img width="407" alt="GitExtensions_2022-12-24_06-35-08" src="https://user-images.githubusercontent.com/10468217/209419837-67116b50-c529-4636-9c3a-20d0e50475a8.png">

**Comment by vbjay** [comment]
That test is to test putty connection over ssh. You are doing a file
protocol. So completely valid result.

**Reply by regs01** [reply]
Result is valid. The problem is when you have multiple remote repositories and you are using Putty. That was the issue about.

**Reply by vbjay** [reply]
The test connection button is for ssh. To validate your ssh configuration is good. It does not validate all protocols.

**Reply by regs01** [reply]
When trying to push into file: it says. Well, yeah, strictly saying - it's not Putty. But it's trying to use it. 
```
GitUI.NBugReports.UserExternalOperationException: SSH key file could not be found
 ---> System.IO.FileNotFoundException: SSH key file could not be found
```

**Reply by vbjay** [reply]
Ah. Now that makes more sense. What you showed earlierooked like it was the test button. 

@RussKie need to adjust the ssh checks only when ssh protocol is being used in your pull request.
